File: lib/features/api_test/api_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:http/http.dart' as http;
import 'package:passdriver/services/api.dart';

void main() {
  group('API Tests', () {
    test('GET request test', () async {
      final response = await http.get(Uri.parse('https://api.passdriver.com/test'));
      expect(response.statusCode, 200);
    });

    test('POST request test', () async {
      final response = await http.post(Uri.parse('https://api.passdriver.com/test'),
          body: {'key': 'value'});
      expect(response.statusCode, 201);
    });
  });
}

File: backend/src/tests/api/api.test.ts
import { Hono } from 'hono';
import { app } from '../hono/app';

describe('API Tests', () => {
  it('GET request test', async () => {
    const res = await app.request('https://api.passdriver.com/test');
    expect(res.status).toBe(200);
  });

  it('POST request test', async () => {
    const res = await app.request('https://api.passdriver.com/test', {
      method: 'POST',
      body: JSON.stringify({ key: 'value' }),
      headers: { 'Content-Type': 'application/json' },
    });
    expect(res.status).toBe(201);
  });
});
